tpot/export_utils.py

```diff
@@ -393,6 +394,24 @@ def replace_function_calls(pipeline_list):
 
 {OUTPUT_DF}['gbc{OPERATOR_NUM}-classification'] = gbc{OPERATOR_NUM}.predict({OUTPUT_DF}.drop('class', axis=1).values)
 """.format(OUTPUT_DF=result_name, OPERATOR_NUM=operator_num, LEARNING_RATE=learning_rate, MAX_FEATURES=max_features, MIN_WEIGHT=min_weight)
+        
+        elif operator_name == '_xg_boosting':
+            
+            learning_rate = min(1., max(float(operator[3]), 0.0001))
+            subsample = min(1., max(0.1, float(operator[4])))            
+            min_child_weight = min(100, max(0, int(operator[5])))
+            max_depth = min(100, max(1, int(operator[6])))            
+
+            if result_name != operator[2]:
+                operator_text += "\n{OUTPUT_DF} = {INPUT_DF}.copy()".format(OUTPUT_DF=result_name, INPUT_DF=operator[2])
+
+            operator_text += """
+# Perform classification with a extreme gradient boosting classifier
+xgbc{OPERATOR_NUM} = XGBClassifier(learning_rate={LEARNING_RATE}, subsample={SUBSAMPLE}, min_child_weight={MIN_WEIGHT}, max_depth={MAX_DEPTH}, n_estimators=500)
+xgbc{OPERATOR_NUM}.fit({OUTPUT_DF}.loc[training_indices].drop('class', axis=1).values, {OUTPUT_DF}.loc[training_indices, 'class'].values)
```

Comment:
Take a look here: http://pt.slideshare.net/DataRobot/gradient-boosted-regression-trees-in-scikitlearn
The parameter tuning section starts on slide 22. The effect of `learning_rate` and `n_estimators` can be seen on slide 25.